### ISSUE TITLE
How to ssh into the UniFi controller?

### ISSUE THREAD

**Issue by dfwsanman** [jacobalberty/unifi-docker#95]
### Host operating system
CentOS 7

### What tag are you using
stable

### What complete docker command or docker-compose.yml do you use to launch the container 
docker run --rm  -p 8222:22 -p 8080:8080 -p 8443:8443 -p 8843:8843 -p 8880:8880 -p 3478:3478/udp -p 6789:6789 -p 10001:10001/udp -e TZ='America/Chicago' -e RUNAS_UID0='false' -e UNIFI_UID='1002' -e UNIFI_GID='1002'  -v ~/unifi:/unifi --net=host --name unifi jacobalberty/unifi:stable

### What do you expect to happen?
ssh -p 8222 host.mydomain.local

### What actually happens?
ssh: connect to host host.mydomain.local port 8222: Connection refused

I'd like to ssh in order to gracefully shutdown.

**Comment by Regner** [comment]
With containers if you really need to get into them you should run `docker exec -it unifi /bin/bash` which, if bash is available in the container, will get you a shell.

**Comment by dfwsanman** [comment]
Thanks, I tried _docker exec -it unifi /bin/bash_ but do not get a bash prompt. Other things like _docker exec -it unifi ls_ do work as expected.

**Comment by phroggster** [comment]
> I'd like to ssh in order to gracefully shutdown.

Pretty sure that [docker-entrypoint.sh](https://github.com/jacobalberty/unifi-docker/blob/master/docker-entrypoint.sh#L5) handles that already when you `docker stop unifi`.

**Comment by jacobalberty** [comment]
That's very odd that `docker exec -it unifi /bin/bash` doesn't give you a prompt, i just tested and it worked fine.

What do you mean by graceful shutdown though? the containers entrypoint script should be handling graceful shutdown of both mongodb and the unifi controller.

**Comment by dfwsanman** [comment]
That would be the plan, but it does not work properly...

from server.log
[2018-01-12 12:56:26,719] <db-server> WARN  system - process is interupted: bin/mongod
[2018-01-12 12:56:26,720] <db-server> INFO  db     - DbServer stopped
[2018-01-12 12:56:26,721] <db-server> WARN  db     - DbServer not shutdown cleanly and need repairing on next startup

from docker
[2018-01-12 12:56:26,790] <docker-entrypoint> WARN: unifi service process ended without being singaled? Check for errors in /unifi/log.

I do not think this is specific to docker, but a general problem with the unifi.init, hence the need to gain access.

**Comment by jacobalberty** [comment]
dfwsanman, how are you planning to gracefully shut it down then? and are you running with --init or no?

**Comment by phroggster** [comment]
Anything helpful in the container's /unifi/log/mongod.log showing what mongod wasn't happy about?

**Comment by dfwsanman** [comment]
As detailed above, I am not using _--init_. I apologize ahead, but I am not very docker literate. When I attempted to use _--init_ I got _unknown flag: --init_

I encountered issues running on cloud key, UniFi randomly becoming not responsive, or losing the config. So I try moving to docker to avoid them, but have now found they are probably more general. The cloud key discussions talk a lot about corruption from unexpected shutdown, and restore from backup.
